src/Grpc.AspNetCore.Server.GrpcClient/GrpcClientServiceExtensions.cs

```diff
@@ -0,0 +1,83 @@
+﻿#region Copyright notice and license
+
+// Copyright 2019 The gRPC Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#endregion
+
+using System;
+using System.Net.Http;
+using Grpc.AspNetCore.Server.GrpcClient;
+using Grpc.Core;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+using Microsoft.Extensions.Http;
+
+namespace Microsoft.Extensions.DependencyInjection
+{
+    public static class GrpcClientServiceExtensions
+    {
+        public static IHttpClientBuilder AddGrpcClient<TClient>(this IServiceCollection services, Action<GrpcClientOptions<TClient>> configureClient) where TClient : ClientBase<TClient>
```

Comment:
You might want another overload that takes a `string name`. Typed clients use `nameof(TClient)` as the identifier for the client, so if you need to register multiple types with the same class name, or register the same class name multiple times you will be in trouble.

Reply:
How do named typed clients work with constructor DI injection?

I'm going to guess that the named configs are usable if IHttpClientFactory is injected into a class and you create an instance from the factory.